src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -673,26 +708,53 @@ private void buildDockerImage() throws PluginExecutionException {
 
     private void startContainer() {
         try {
-            containerID = execDockerCmd(getContainerCommand(), 30);
-            info("docker container: " + containerID);
-        } catch (RuntimeException r) {
-            error("Error starting container: " + r.getMessage());
-            throw r;
+            String startContainerCommand = getContainerCommand();
+            debug("startContainer, cmd="+startContainerCommand);
+
+            ProcessBuilder processBuilder = new ProcessBuilder();
+            processBuilder.command(getCommandTokens(startContainerCommand));
+            processBuilder.redirectOutput(Redirect.INHERIT);
+            dockerRunProcess = processBuilder.start();
+            dockerRunProcess.waitFor();
+            if (dockerRunProcess.exitValue() != 0) {
+                info("Error running docker command, return value=" + dockerRunProcess.exitValue());
+                // read messages from standard err
+                char[] d = new char[1023];
+                new InputStreamReader(dockerRunProcess.getErrorStream()).read(d);
+                String errorMessage = new String(d).trim() + " RC=" + dockerRunProcess.exitValue();
+                throw new RuntimeException(errorMessage);
+            }
+        } catch (IOException e) {
+            error("Error starting container: " + e.getMessage());
+            throw new RuntimeException(e);
+        } catch (InterruptedException e) {
+            error("Thread was interrupted while starting the container: " + e.getMessage());
         }
     }
 
+    private String[] getCommandTokens(String command) {
+        StringTokenizer stringTokenizer = new StringTokenizer(command);
+        String[] commandTokens = new String[stringTokenizer.countTokens()];
+        for (int i = 0; stringTokenizer.hasMoreTokens(); i++) {
+            commandTokens[i] = stringTokenizer.nextToken();
+        }
+        return commandTokens;
+    }
+
     private void stopContainer() {
         try {
-            info("stopping docker container: " + containerID);
-            if (containerID != null) {
-                String s = execDockerCmd("docker stop " + containerID, 30);
+            info("Stopping docker container...");
+            if (dockerRunProcess != null) {
+                // Emulate docker stop with SIGTERM, wait up to 10 seconds, then SIGKILL
+                dockerRunProcess.destroy();
+                dockerRunProcess.waitFor(10, TimeUnit.SECONDS);
```

Comment:
If I kill the container it sometimes takes about 30s to disappear. Could you make this timeout longer?

Reply:
`docker stop`'s default timeout is 10 seconds though. https://docs.docker.com/engine/reference/commandline/stop/

Reply:
Ok, looks good.

Reply:
I realized that waitFor() returned immediately once I changed the implementation in the container thread to use a reader to log its content to Maven/Gradle instead of setting `processBuilder.redirectOutput(Redirect.INHERIT);`.  This caused server restart to fail since that meant `stopContainer()` was no longer synchronous.  So I changed it now to use `docker ps` and then `docker stop`, which is synchronous.  Can you please take another look?